pygmt/params/base.py

```diff
@@ -0,0 +1,97 @@
+"""
+Base class for common parameters shared in PyGMT.
+"""
+
+
+class BaseParam:
```

Comment:
Probably should make this an [AbstractBaseClass](https://docs.python.org/3/library/abc.html) (was debating between ABC or a [Protocol](https://peps.python.org/pep-0544/)).

```suggestion
from abc import ABC, abstractmethod, abstractproperty


class BaseParam(ABC):
```

Reply:
I've applied the suggestions but also made some changes in c6bd16f1b90144f6d4b267037f58b1439a2e1161. The main changes are:

- `abstractproperty` is deprecated https://docs.python.org/3/library/abc.html#abc.abstractproperty. Now use `@property` and `@abstracmethod`. 
- The `@abstractmethod` means subclasses must implement the method/property, so the `NotImplementedError` is not needed and can be removed.
- The `_validate` method is meant to be optional, so we can't use `@abstractmethod`. Then ruff reports the [`B027`](https://docs.astral.sh/ruff/rules/empty-method-without-abstract-decorator/) error, and it's ignored by adding the `noqa` flag.